### ISSUE TITLE
Improve `invalid manifest` error wording when manifest can't be parsed.

### ISSUE THREAD

**Issue by MaxDesiatov** [apple/swift-package-manager#5886]
### Description

As [suggested on Swift Forums](https://forums.swift.org/t/missing-swift-tools-version-specification-build-fail/61051/10) `error: 'manifest': Invalid manifest` is not a helpful message. We should explain a details where exactly the error has occurred and what users can do about it to fix it.

### Expected behavior

When a package manifest can't be parsed, SwiftPM should tell the user what exactly it expects the manifest to be and what it should contain.

### Actual behavior

Only `error: 'manifest': Invalid manifest` is displayed when the package manifest can't be parsed.

### Steps to reproduce

1. Go to an empty directory or a directory that doesn't have `Package.swift` in it.
2. Run `echo '// swift-tools-version: 5.7.1\nprint("hello")' > Package.swift `
3. Run `swift build`

### Swift Package Manager version/commit hash

5.7.1

### Swift & OS version (output of `swift --version && uname -a`)

swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
Darwin MacBook-Pro.local 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000 arm64

**Comment by neonichu** [comment]
Agreed, seems bad. I think what this actually means is that we couldn't get a parseable JSON response, but no compilation errors. In this case, because there's actually no instance of `Package` being created.
